azure-pipelines.yml

```diff
@@ -0,0 +1,82 @@
+parameters:
+  - name: terraformVersion
+    type: string
+    default: "0.13.4"
+  - name: sendgridEnvironments
+    type: object
+    default:
+    - environment: nonprod
+      serviceConnection: dts-management-nonprod-intsvc
+      rg: mgmt-state-store-nonprod
+      location: uksouth
+      storage_account: dtsmgmtstatestorenonprod
+      container: mgmtstatestorecontainernonprod
+
+    - environment: prod
+      serviceConnection: dts-management-prod-intsvc
+      rg: mgmt-state-store-prod
+      location: uksouth
+      storage_account: dtsmgmtstatestoreprod
+      container: mgmtstatestorecontainerprod
+
+variables:
+  isMain: $[eq(variables['Build.SourceBranch'], 'refs/heads/master')]
+
+pr:
+- Integration
+
+pool:
+  name: Azure Pipelines
+  demands: azureps
+  vmImage: 'windows-latest'
+
+
+stages:
+  - ${{ each environment in parameters.sendgridEnvironments }}:
+    - stage: ${{ environment.environment }}
+      jobs:
+        - job: TestAndPlan
+          steps:     
+          - task: TerraformInstaller@0
+            displayName: Terraform install
+            inputs:
+              terraformVersion: ${{ parameters.terraformVersion }}
+
+          - task: TerraformCLI@0
+            displayName: Terraform initialize
+            inputs:
+              command: init
+              workingDirectory: $(System.DefaultWorkingDirectory)/terraform
+              backendType: azurerm
+              ensurebackend: true
+              backendServiceArm: ${{ environment.serviceConnection }}
+              backendAzureRmResourceGroupName: ${{ environment.rg }}
```

Comment:
should be calculated based on a pattern

Reply:
unclear on what is meant here but parameters now stripped down to environments only and others changed to patterned variables.